### ISSUE TITLE
2 test failures while building for debian with Python 3.7

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#2689]
#### What is the problem?
We need to add 3.7 testing, at least for a single matrix run or may be (also) for appveyor to spread the loa

```
======================================================================
FAIL: datalad.support.tests.test_network.test_url_samples
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/datalad-0.10.2/.pybuild/cpython3_3.7_datalad/build/datalad/support/tests/test_network.py", line 249, in test_url_samples
    _check_ri("file:///~/path/sp1", URL, localpath='/~/path/sp1', scheme='file', path='/~/path/sp1', exact_str=False)
  File "/build/datalad-0.10.2/.pybuild/cpython3_3.7_datalad/build/datalad/support/tests/test_network.py", line 139, in _check_ri
    assert_in('Parsed version of', cml.out)
AssertionError: 'Parsed version of' not found in ''

======================================================================
FAIL: datalad.support.tests.test_network.test_get_local_file_url_linux
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/datalad-0.10.2/.pybuild/cpython3_3.7_datalad/build/datalad/tests/utils.py", line 825, in newfunc
    return func(*args, **kwargs)
  File "/build/datalad-0.10.2/.pybuild/cpython3_3.7_datalad/build/datalad/support/tests/test_network.py", line 406, in test_get_local_file_url_linux
    eq_(get_local_file_url('/a~'), 'file:///a%7E')
AssertionError: 'file:///a~' != 'file:///a%7E'
```

**Comment by yarikoptic** [comment]
testing locally with ssh tests lead to the whole nose crash without proper completion seems to me due to vcr:
```
datalad.downloaders.tests.test_s3.test_s3_download_basic('s3://datalad-test0-versioned/2versions-nonversioned1.txt?versionId=null', 'version2', 'version1') ... Versions: appdirs=1.4.3 boto=2.44.0 cmd:annex=6.20180521+gitg0e3bebd71-1~ndall+1 cmd:git=2.17.0 cmd:system-git=2.18.0 cmd:system-ssh=7.6p1 exifread=2.1.2 git=2.1.8 gitdb=2.0.3 humanize=0.5.1 iso8601=0.1.11 msgpack=0.5.1 mutagen=1.38 patoolib=1.12 requests=2.18.4 six=1.11.0 tqdm=4.19.5 wrapt=1.9.0
Obscure filename: str=b' "\';a&b&c\xce\x94\xd0\x99\xd7\xa7\xd9\x85\xe0\xb9\x97\xe3\x81\x82 `| ' repr=' "\';a&b&cΔЙקم๗あ `| '
Encodings: default='utf-8' filesystem='utf-8' locale.prefered='UTF-8'
Environment: LANG='en_US.utf8' PATH='/home/yoh/proj/datalad/datalad/venvs/dev3.7/bin:/home/yoh/gocode/bin:/home/yoh/gocode/bin:/home/yoh/bin:/home/yoh/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/snap/bin:/sbin:/usr/sbin:/usr/local/sbin' GIT_PAGER='less --no-init --quit-if-one-screen' GIT_PYTHON_GIT_EXECUTABLE='/usr/lib/git-annex.linux/git'
SKIP: This test requires known credentials for datalad-test-s3
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/vcr/cassette.py", line 137, in _handle_generator
    to_yield = coroutine.send(to_send)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 251, in generate
    for test in g():
RuntimeError: generator raised StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3/dist-packages/nose/__main__.py", line 8, in <module>
    run_exit()
  File "/usr/lib/python3/dist-packages/nose/core.py", line 121, in __init__
    **extra_args)
  File "/usr/lib/python3.7/unittest/main.py", line 101, in __init__
    self.runTests()
  File "/usr/lib/python3/dist-packages/nose/core.py", line 207, in runTests
    result = self.testRunner.run(self.test)
  File "/usr/lib/python3/dist-packages/nose/core.py", line 62, in run
    test(result)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 217, in run
    for test in self._tests:
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 372, in _get_wrapped_tests
    for test in self._get_tests():
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 99, in _get_tests
    for test in self.test_generator:
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 261, in generate
    address=test_address(generator))
  File "/usr/lib/python3/dist-packages/nose/util.py", line 437, in test_address
    raise TypeError("I don't know what %s is (%s)" % (test, t))
TypeError: I don't know what <function test_s3_download_basic at 0x7f75039582f0> is (<class 'FunctionWrapper'>)
DATALAD_TESTS_SSH=1 python -m nose -s -v datalad 2>&1  610.96s user 264.03s system 91% cpu 16:00.69 total
tee tests-3.7.log  0.01s user 0.02s system 0% cpu 16:00.69 total
```

**Comment by kyleam** [comment]
AFAICT the fix was merged in with 60ca887b8 (BF(RF): do not quote ~ in URLs, 2018-07-09).  At any rate we now have py3.7 in the build, and these tests are passing.
